src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/ByteArrayMemoryBlock.cs

```diff
@@ -25,15 +25,15 @@ internal ByteArrayMemoryBlock(ByteArrayMemoryProvider provider, int start, int s
 
         public override void Dispose()
         {
-            _provider = null;
+            _provider = null!;
         }
 
-        public unsafe override byte* Pointer => _provider.Pointer + _start;
+        public unsafe override byte* Pointer => _provider!.Pointer + _start;
```

Comment:
`_provider` is non-nullable why do we need a `!`?

Reply:
Good catch. I had initially annotated as nullable because of the Dispose method above.